File: monkey-web/src/constant/FormTypeEnum.js

let FormTypeEnum = createEnum({
    FORM_TYPE_ALL: ['-1', "全部"],
    FORM_TYPE_FREE: ['1', '免费'],
    FORM_TYPE_VIP: ['2', '会员免费'],
    FORM_TYPE_TOLL: ["3", "收费"],
    FORM_TYPE_COMMEND: ["4", "官方推荐"]
});

function createEnum(definition) {
    const valueMap = {};
    const descMap = {};
    for (const key of Object.keys(definition)) {
        const [value, desc] = definition[key];
        valueMap[key] = value;
        descMap[value] = desc;
    }
    return {
        ...valueMap,
        getCode(key) {
            return (definition[key] && definition[key][1]) || '无';
        },
        getMsg(value) {
            return descMap[value] || '无';
        }
    }
}


export default FormTypeEnum

File: monkey-common/common-utils/src/main/java/com/monkey/monkeyUtils/constants/FormTypeEnum.java
package com.monkey.monkeyUtils.constants;

/**
 * @author: wusihao
 * @date: 2023/7/27 16:52
 * @version: 1.0
 * @description:
 */
public enum FormTypeEnum {
    FORM_TYPE_ALL(-1L, "全部"),
    FORM_TYPE_FREE(1L, "免费"),
    FORM_TYPE_VIP(2L, "会员免费"),
    FORM_TYPE_TOLL(3L, "收费"),
    FORM_TYPE_COMMEND(4L, "官方推荐")
            ;

    private Long code;
    private String msg;

    FormTypeEnum(Long code, String msg) {
        this.code = code;
        this.msg = msg;
    }

    public Long getCode() {
        return code;
    }

    public String getMsg() {
        return msg;
    }

    //获取指定值枚举类
    public static FormTypeEnum getFormTypeEnum(Long code) {
        //code为null
        if (null == code) {
            return FormTypeEnum.FORM_TYPE_ALL;
        }
        FormTypeEnum[] values = FormTypeEnum.values();
        for (FormTypeEnum value : values) {
            if (value.code.equals(code)) {
                return value;
            }
        }
        //没找到、
        return FormTypeEnum.FORM_TYPE_ALL;
    }
}
